src/pe_source/intelx_identity.py

```diff
@@ -0,0 +1,289 @@
+"""Collect Intelx credential leak data."""
+# Standard Python Libraries
+import datetime
+import logging
+import sys
+import time
+
+# Third-Party Libraries
+import numpy as np
+import pandas as pd
+import requests
+
+from .data.pe_db.config import get_params
+from .data.pe_db.db_query_source import (
+    connect,
+    get_data_source_uid,
+    get_intelx_breaches,
+    get_orgs,
+    insert_intelx_breaches,
+    insert_intelx_credentials,
+    org_root_domains,
+)
+
+# Calculate Datetimes for collection period
+TODAY = datetime.date.today()
+DAYS_BACK = datetime.timedelta(days=16)
+START_DATE = (TODAY - DAYS_BACK).strftime("%Y-%m-%d %H:%M:%S")
+END_DATE = TODAY.strftime("%Y-%m-%d %H:%M:%S")
+
+
+section = "intelx"
+params = get_params(section)
+api_key = params[0][1]
+
+LOGGER = logging.getLogger(__name__)
+
+
+class IntelX:
+    """Fetch IntelX data."""
+
+    def __init__(self, orgs_list):
+        """Initialize IntelX class."""
+        LOGGER.info("Initialized IntelX")
+        self.orgs_list = orgs_list
+
+    def run_intelx(self):
+        """Run IntelX api calls."""
+        orgs_list = self.orgs_list
+
+        pe_orgs = get_orgs()
+        for pe_org in pe_orgs:
+            cyhy_org_id = pe_org["cyhy_db_name"]
+            pe_org_uid = pe_org["org_uid"]
+
+            # Verify the org is in the list of orgs to scan
+            if cyhy_org_id in orgs_list or orgs_list == "all":
+                if self.get_credentials(cyhy_org_id, pe_org_uid) == 1:
+                    LOGGER.error("Failed to get credentials for %s", cyhy_org_id)
+
+    def get_credentials(self, cyhy_org_id, pe_org_uid):
+        """Get credentials for a provided org."""
+        LOGGER.info("Fetching credential data for %s.", cyhy_org_id)
+        source_uid = get_data_source_uid("IntelX")
+        try:
+            conn = connect()
+            roots_df = org_root_domains(conn, pe_org_uid)
+            LOGGER.info(f"Got roots for {cyhy_org_id}")
+        except Exception as e:
+            LOGGER.error("Failed fetching root domains for %s", cyhy_org_id)
+            LOGGER.error(e)
+            return 1
+
+        leaks_json = self.find_credential_leaks(
+            roots_df["root_domain"].values.tolist(), START_DATE, END_DATE
+        )
+        if len(leaks_json) < 1:
+            LOGGER.info(f"No credentials found for {cyhy_org_id}")
```

Comment:
We avoid using f-strings in logging statements - see [here](https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/) for some good reasons why:
```suggestion
            LOGGER.info("No credentials found for %s", cyhy_org_id)
```

I noticed a bunch other instances of using f-strings in logging messages in this PR - please modify them as well (I don't want to flag them all now, but I'll check for them before I give my approval).

Reply:
@dav3r  other instances fixed in the following commit: [2b9e3ea](https://github.com/cisagov/pe-reports/pull/414/commits/2b9e3eaaddca5609045a8330d61705c849488b08)